src/index.ts

```diff
@@ -35,73 +36,128 @@ let traceAgent: StackdriverTracer;
 /**
  * Normalizes the user-provided configuration object by adding default values
  * and overriding with env variables when they are provided.
- * @param projectConfig The user-provided configuration object. It will not
+ * @param userConfig The user-provided configuration object. It will not
  * be modified.
  * @return A normalized configuration object.
  */
-function initConfig(projectConfig: Forceable<Config>):
-    Forceable<NormalizedConfig> {
-  // `|| undefined` prevents environmental variables that are empty strings
-  // from overriding values provided in the config object passed to start().
-  const envConfig = {
-    logLevel: Number(process.env.GCLOUD_TRACE_LOGLEVEL) || undefined,
-    projectId: process.env.GCLOUD_PROJECT || undefined,
-    serviceContext: {
-      service:
-          process.env.GAE_SERVICE || process.env.GAE_MODULE_NAME || undefined,
-      version: process.env.GAE_VERSION || process.env.GAE_MODULE_VERSION ||
-          undefined,
-      minorVersion: process.env.GAE_MINOR_VERSION || undefined
-    }
-  };
-
-  let envSetConfig: Config = {};
+function initConfig(userConfig: Forceable<Config>): Forceable<TopLevelConfig> {
+  // Helper function which gets the last non-null/undefined value among its
+  // arguments. The first argument must therefore be guaranteed to be
+  // non-null/undefined.
+  const lastOf =
+      <T>(defaultValue: T, ...otherValues: Array<T|null|undefined>): T => {
+        for (let i = otherValues.length - 1; i >= 0; i--) {
+          // tslint:disable:no-any
+          if (otherValues[i] != null &&
```

Comment:
Is the `!=` intentional or did you want a `!==`?

Reply:
It was intentional. That being said, I've separated it out into `!== undefined` and `!== null` for readability.